kornia/enhance/adjust.py

```diff
@@ -529,6 +529,76 @@ def adjust_brightness_accumulative(image: Tensor, factor: Union[float, Tensor],
     return img_adjust
 
 
+def adjust_sigmoid(image: Tensor, cutoff: float = 0.5, gain: float = 10, inv: bool = False) -> Tensor:
+    """Adjust sigmoid correction on the input image tensor.
+
+    The input image is expected to be in the range of [0, 1].
+
+    Reference:
+    [1]: Gustav J. Braun, "Image Lightness Rescaling Using Sigmoidal Contrast Enhancement Functions",
+        http://markfairchild.org/PDFs/PAP07.pdf
+
+    Args:
+        image: Image to be adjusted in the shape of :math:`(*, H, W)`.
+        cutoff: The cutoff of sigmoid function.
+        gain: The multiplier of sigmoid function.
+        inv: If is set to True the function will return the inverse sigmoid correction.
+
+    Returns:
+         Adjusted tensor in the shape of :math:`(*, H, W)`.
+
+    Example:
+        >>> x = torch.ones(1, 1, 2, 2)
+        >>> adjust_sigmoid(x, gain=0)
+        tensor([[[[0.5000, 0.5000],
+                  [0.5000, 0.5000]]]])
+    """
+    KORNIA_CHECK_IS_TENSOR(image, "Expected shape (*, H, W)")
+
+    if inv:
```

Comment:
I think it would make sense to have `adjust_sigmoid` and `adjust_sigmoid_inverse`. I'm not a fan of free functions that by a flag behaves totally different.

Reply:
Not sure about that, as we would have four functions instead of 2. I understand that you are pointing to the Clean Code suggestion, but personally, I for experiments prefer flag. I can modify that if you think that it would be better.

Reply:
i'd like to have all this set of low level operators with very specific naming of what they do and possibly with proper references. We had similar discussion during the contrast/brightness design specially because there's a trend of just copying behaviors from other libraries that don't follow the theoretical definition. In that case we decided doing something similar to here with [`adjust_contrast_with_mean_subtraction`](https://github.com/kornia/kornia/blob/master/kornia/enhance/adjust.py#L357) /cc @shijianjian @ducha-aiki